redundant-bmc/src/manager.cpp

```diff
@@ -110,12 +166,66 @@ Role Manager::determineRole()
         roleInfo.error = ErrorCase::internalError;
     }
 
-    if (roleInfo.error != role_determination::ErrorCase::noError)
+    // TODO, probably: Create an error log if passive due to an error
+
+    return roleInfo;
+}
+
+// NOLINTBEGIN
+sdbusplus::async::task<std::optional<role_determination::RoleInfo>>
+    Manager::determinePassiveRoleIfRequired()
+// NOLINTEND
+{
+    using namespace role_determination;
+
+    // An unprovisioned BMC cannot be active.
+    if (!services->getProvisioned())
+    {
+        lg2::info("Role = passive because BMC not provisioned");
+        co_return RoleInfo{Role::Passive, ErrorCase::notProvisioned};
+    }
+
+    // The sibling service must be up and running.
+    if (!sibling->getInterfacePresent())
     {
-        // TODO: Create an error log
+        auto state = co_await services->getUnitState(Sibling::unitName);
```

Comment:
I'm not thrilled about hardcoding the service name.  The alternative would be if the rbmc-cfamd's object manager interface is on D-Bus which would also have to hardcode something.

Reply:
will leave as is for now.  can revisit in the future if necessary.